src/Compilers/CSharp/Portable/Compilation/CSharpCompilation.cs

```diff
@@ -1543,35 +1543,47 @@ internal bool DeclaresTheObjectClass
 
         internal EntryPoint? GetEntryPointAndDiagnostics(CancellationToken cancellationToken)
```

Comment:
Do we ever expect this function to return `null` any more now that we use `EntryPoint.None` to indicate "we looked for an entry point and didn't find one"?

Reply:
> Do we ever expect this function to return null any more now that we use EntryPoint.None to indicate "we looked for an entry point and didn't find one"?

No, I'll remove the ``?``` from the return type. Good catch.

---
In reply to: [416818038](https://github.com/dotnet/roslyn/pull/43738#discussion_r416818038) [](ancestors = 416818038)

Reply:
Consider also removing the null check on the return value in `MethodCompiler.GetEntryPoint`.